aptos-move/framework/src/natives/cryptography/ristretto255_point.rs

```diff
@@ -0,0 +1,624 @@
+}
+
+pub(crate) fn native_new_point_from_sha512(
+    gas_params: &GasParameters,
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert_eq!(ty_args.len(), 0);
+    assert_eq!(args.len(), 1);
+
+    let point_context = context.extensions().get::<NativeRistrettoPointContext>();
+    let mut point_data = point_context.point_data.borrow_mut();
+
+    let bytes = pop_arg!(args, Vec<u8>);
+
+    let cost = gas_params.base_cost
+        + gas_params.point_from_64_uniform_bytes_cost * NumArgs::one()
+        + gas_params.sha512_per_hash_cost * NumArgs::one()
+        + gas_params.sha512_per_byte_cost * NumBytes::new(bytes.len() as u64);
+
+    let result_handle = point_data.add_point(RistrettoPoint::hash_from_bytes::<Sha512>(&bytes));
+
+    Ok(NativeResult::ok(cost, smallvec![Value::u64(result_handle)]))
+}
+
+pub(crate) fn native_new_point_from_64_uniform_bytes(
+    gas_params: &GasParameters,
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert_eq!(ty_args.len(), 0);
+    assert_eq!(args.len(), 1);
+
+    let point_context = context.extensions().get::<NativeRistrettoPointContext>();
+    let mut point_data = point_context.point_data.borrow_mut();
+
+    let slice = pop_64_byte_slice(&mut args)?;
+
+    let cost = gas_params.base_cost + gas_params.point_from_64_uniform_bytes_cost * NumArgs::one();
+
+    let result_handle = point_data.add_point(RistrettoPoint::from_uniform_bytes(&slice));
+
+    Ok(NativeResult::ok(cost, smallvec![Value::u64(result_handle)]))
+}
+
+pub(crate) fn native_multi_scalar_mul(
+    gas_params: &GasParameters,
+    context: &mut NativeContext,
+    mut ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert_eq!(ty_args.len(), 2);
+    assert_eq!(args.len(), 2);
+
+    let point_context = context.extensions().get::<NativeRistrettoPointContext>();
+    let mut point_data = point_context.point_data.borrow_mut();
+
+    let mut cost = gas_params.base_cost;
+
+    let scalar_type = ty_args.pop().unwrap();
+    let point_type = ty_args.pop().unwrap();
+
+    let scalars_ref = pop_arg!(args, VectorRef);
+    let points_ref = pop_arg!(args, VectorRef);
+
+    let num = scalars_ref.len(&scalar_type)?.value_as::<u64>()? as usize;
+
+    // parse scalars
+    let mut scalars = Vec::with_capacity(num);
+    for i in 0..num {
+        let move_scalar = scalars_ref.borrow_elem(i, &scalar_type)?;
+        let scalar = scalar_from_struct(move_scalar)?;
+
+        scalars.push(scalar);
+    }
+
+    // parse points
+    let mut points = Vec::with_capacity(num);
+    for i in 0..num {
+        let move_point = points_ref.borrow_elem(i, &point_type)?;
+        let point_handle = get_point_handle_from_struct(move_point)?;
+
+        points.push(point_data.get_point(&point_handle));
+    }
```

Comment:
**TODO:** Add a cost variable for this potentially-expensive argument parsing of a Move `&vector<Scalar>` and `&vector<RistrettoPoint>` into Rust vectors.

@vgao1996 does this seem reasonable to you?